### PR TITLE
Refactors for track registry

### PR DESCRIPTION
**Pull request by liach**
Cleanup after api pull.
Seems it cannot grab the correct submodule, so it fails to build.
See https://github.com/CovertJaguar/Railcraft-API/pull/18

### CODE REVIEW

**Comment by liach** [comment]
As long as you merge the api one first and rebuild, this will pass.

File: src/main/java/mods/railcraft/common/util/misc/Game.java [hunk 1]
```diff
@@ -33,17 +32,13 @@
...
+            obfuscated = MinecraftServer.class.getDeclaredField("worldServers") == null;
+        } catch (NoSuchFieldException absence) {
+            obfuscated = true;
         }
-        try {
-            worldObjFound = Entity.class.getDeclaredField("worldObj") != null;
-        } catch (NoSuchFieldException | SecurityException ignored) {
-        }
-        OBFUSCATED = !worldFound && !worldObjFound;
+        OBFUSCATED = obfuscated;
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/util/misc/Game.java]: Why are we changing this shit again? I just got done testing it in all possible environments. -.-

**Review by CovertJaguar — CHANGES_REQUESTED** [review]

File: src/main/java/mods/railcraft/common/util/misc/Game.java (hunk 1 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/util/misc/Game.java]: Cuz this is done after merging. You should consider this as a merge conflict, and I will fix it too,

**Review by CovertJaguar — APPROVED** [review]

**Comment by liach** [comment]
Let me fix submodules.

**Comment by liach** [comment]
Guess I will merge this one first

**Pull request merged**